### PR TITLE
Support volume mount subpaths and read-only flag

### PR DESCRIPTION
**Pull request by spadgett**
Show subpath and read-only values for volume mounts. Allow users to set these fields on the add storage page.

### CODE REVIEW

File: app/views/attach-pvc.html [hunk 1]
```diff
@@ -154,6 +180,18 @@ <h2 class="text-center">No persistent volume claims.</h2>
...
                       </div>
 
+                      <div class="form-group">
+                        <div class="checkbox">
+                          <label>
+                            <input type="checkbox" ng-model="attach.readOnly" aria-describedby="read-only-help">
+                            Read only
+                          </label>
+                          <div id="read-only-help" class="help-block">
+                            Mount the volume read-only.
```
Comment (jwforres) [inline app/views/attach-pvc.html]: Mount the volume as read-only.

File: app/views/attach-pvc.html [hunk 2]
```diff
@@ -113,6 +113,32 @@ <h2 class="text-center">No persistent volume claims.</h2>
...
+                            ng-model="attach.subPath"
+                            placeholder="example: application/resources"
+                            ng-pattern="RELATIVE_PATH_PATTERN"
+                            autocorrect="off"
+                            autocapitalize="off"
+                            spellcheck="false"
+                            aria-describedby="sub-path-help">
+                        <div id="sub-path-help" class="help-block">
+                          Optional path within the volume from which the container's volume should
+                          be mounted. Defaults to the volume's root.
```
Comment (jwforres) [inline app/views/attach-pvc.html]: Maybe:
Optional path within the volume from which it will be mounted into the container. Defaults to the volume's root.

**Comment by jwforres** [comment]
also this needs a rebase

**Comment by spadgett** [comment]
@jwforres thanks, updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**